types/mongoose/index.d.ts

```diff
@@ -83,9 +84,21 @@ declare module "mongoose" {
   // We can use TypeScript Omit once minimum required TypeScript Version is above 3.5
   type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
 
+  type NonFunctionPropertyNames<T> = {
+    [K in keyof T]: T[K] extends Function ? never : K;
+  }[keyof T];
+  
+  type NonFunctionProperties<T> = Pick<T, NonFunctionPropertyNames<T>>;
+  
+  type DeepNonFunctionProperties<T> = {
+    [K in keyof NonFunctionProperties<T>]: T[K] extends object ? DeepNonFunctionProperties<T[K]> : T[K]
+  };
+
   /* Helper type to extract a definition type from a Document type */
   type DocumentDefinition<T> = Omit<T, Exclude<keyof Document, '_id'>>;
 
+  type CreateDocumentDefinition<T> = DeepNonFunctionProperties<DocumentDefinition<T>>;
```

Comment:
Yes, but `FilterQuery` and `UpdateQuery` mark all properties optional, so any functions in the interface is there but not required.

For `Create` it can pop up as required, if it's required in the interface.

I know that in theory this is a code smell when the interface defines a function in it, when used for the schema. 

But... in actual usage it's pretty common to reuse the type of a class as the type of the schema, so this just makes it nice and removes all functions from the type used for creation.